src/routers/handlers/enter_your_details/index.ts

```diff
@@ -73,7 +74,7 @@ export class EnterYourDetailsHandler extends GenericHandler<EnterYourDetailsView
         }
 
         setPresenterAccountDetails(req, details);
-        return { redirect: PrefixedUrls.CHECK_DETAILS };
+        return { redirect: urlWithLang(PrefixedUrls.CHECK_DETAILS, req.query.lang) };
```

Comment:
I'm not sure I like this, it's going to lead to a lot of repeated code. Could we look at changing how prefixed urls works so that it includes the lang automatically?

Reply:
Maybe if it looked like this: 
```suggestion
        return { redirect: getUrl( PageKey.CHECK_DETAILS, req.query.lang) };
```
Make it so you have to include the lang to get a url.